wheels/install_deps_linux.sh

```diff
@@ -0,0 +1,146 @@
+    curl -SL https://github.com/xianyi/OpenBLAS/archive/v${openblas_version}.tar.gz -o ${openblas_pkg}
+fi
+
+echo "Building OpenBLAS..."
+
+rm -rf ${openblas_dir}
+tar xzf ${openblas_pkg} \
+    && pushd ${openblas_dir} >/dev/null 2>&1 \
+    && make USE_OPENMP=1 NO_SHARED=1 \
+    MAKE_NB_JOBS=${MAKEJ} \
+    CC="${CC}" FC="${FC}" DYNAMIC_ARCH=1 \
+    COMMON_OPT="${CFLAGS}" FCOMMON_OPT="${FCFLAGS}" \
+    LDFLAGS="-fopenmp -lm" \
+    && make NO_SHARED=1 PREFIX="${PREFIX}" install \
+    && popd >/dev/null 2>&1
+
+# Install FFTW
+
+fftw_version=3.3.8
+fftw_dir=fftw-${fftw_version}
+fftw_pkg=${fftw_dir}.tar.gz
+
+echo "Fetching FFTW..."
+
+if [ ! -e ${fftw_pkg} ]; then
+    curl -SL http://www.fftw.org/${fftw_pkg} -o ${fftw_pkg}
+fi
+
+echo "Building FFTW..."
+
+rm -rf ${fftw_dir}
+tar xzf ${fftw_pkg} \
+    && pushd ${fftw_dir} >/dev/null 2>&1 \
+    && CC="${CC}" CFLAGS="${CFLAGS}" \
+    ./configure \
+    --enable-threads \
+    --enable-static \
+    --disable-shared \
+    --prefix="${PREFIX}" \
+    && make -j ${MAKEJ} \
+    && make install \
+    && popd >/dev/null 2>&1
+
+# Install libaatm
+
+aatm_version=1.0.9
+aatm_dir=libaatm-${aatm_version}
+aatm_pkg=${aatm_dir}.tar.gz
+
+echo "Fetching libaatm..."
+
+if [ ! -e ${aatm_pkg} ]; then
+    curl -SL "https://github.com/hpc4cmb/libaatm/archive/${aatm_version}.tar.gz" -o "${aatm_pkg}"
+fi
+
+echo "Building libaatm..."
+
+rm -rf ${aatm_dir}
+tar xzf ${aatm_pkg} \
+    && pushd ${aatm_dir} >/dev/null 2>&1 \
+    && mkdir -p build \
+    && pushd build >/dev/null 2>&1 \
+    && cmake \
+    -DCMAKE_C_COMPILER="${CC}" \
+    -DCMAKE_CXX_COMPILER="${CXX}" \
+    -DCMAKE_C_FLAGS="${CFLAGS}" \
+    -DCMAKE_CXX_FLAGS="${CXXFLAGS}" \
+    -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+    -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
+    .. \
+    && make -j ${MAKEJ} install \
+    && popd >/dev/null 2>&1 \
+    && popd >/dev/null 2>&1
+
+# Install SuiteSparse
+
+ssparse_version=5.7.2
+ssparse_dir=SuiteSparse-${ssparse_version}
+ssparse_pkg=${ssparse_dir}.tar.gz
+
+echo "Fetching SuiteSparse..."
+
+if [ ! -e ${ssparse_pkg} ]; then
+    curl -SL https://github.com/DrTimothyAldenDavis/SuiteSparse/archive/v${ssparse_version}.tar.gz -o ${ssparse_pkg}
+fi
+
+echo "Building SuiteSparse..."
+
+rm -rf ${ssparse_dir}
+tar xzf ${ssparse_pkg} \
+    && pushd ${ssparse_dir} >/dev/null 2>&1 \
+    && patch -p1 < "${topdir}/suitesparse.patch" \
+    && make library JOBS=${MAKEJ} \
+    CC="${CC}" CXX="${CXX}" \
+    CFLAGS="${CFLAGS}" CXXFLAGS="${CXXFLAGS}" AUTOCC=no \
+    GPU_CONFIG="" CFOPENMP="${OPENMP_CXXFLAGS}" BLAS="-lopenblas -fopenmp -lm" \
+    && make static JOBS=${MAKEJ} \
+    CC="${CC}" CXX="${CXX}" \
+    CFLAGS="${CFLAGS}" CXXFLAGS="${CXXFLAGS}" AUTOCC=no \
+    GPU_CONFIG="" CFOPENMP="${OPENMP_CXXFLAGS}" BLAS="-lopenblas -fopenmp -lm" \
+    && cp -a ./include/* "${PREFIX}/include/" \
+    && find . -name "*.a" -exec cp -a '{}' "${PREFIX}/lib/" \; \
+    && popd >/dev/null 2>&1
+
```

Comment:
there is so much building from source, isn't there another way? maybe do 1 docker container and update it once in a while? or conda install those?

Reply:
For linux we could do that by taking the cibuildwheel container installing our stuff, and pushing that new container to dockerhub.  For MacOS, I don't think we can do that...

Reply:
Re: conda.  I'm not sure we can mix conda-installed libraries with the system ones.  Conda uses its own compiler toolchain which acts like a cross compiler.  Of course it would also require installing a miniconda base system here and potentially debugging future incompatibilities.  The downside to building these things from source is the time it takes.  The upside is that we can control how they are built and link them statically into our compiled extension.  So our only external dependencies are libc, libm, etc

Reply:
I will leave this as-is for now, since it is working.  If we find a way to reliably install suitesparse / openblas from a package manager then this can be a new PR to this file in the future.